### ISSUE TITLE
Inline/extract refactorings should warn on left-hand-side of deconstruction

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#19047]
This relates to the inferred tuple names feature.
Some of the refactorings should add annotations (`ConflictAnnotation`) to indicate this is likely not correct.

```C#
        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary)]
        public async Task ExplicitTupleNameAdded_DeconstructionDeclaration()
        {
            var code = @"
class C
{
    static int y = 1;
    void M()
    {
        int [||]i = C.y;
        var t = ((i, (i, _)) = (1, (i, 3)));
    }
}";

            var expected = @"
class C
{
    static int y = 1;
    void M()
    {
        var t = (((int)C.y, ((int)C.y, _)) = (1, (C.y, 3)));
    }
}";
            await TestInRegularAndScriptAsync(code, expected, ignoreTrivia: false);
        }

        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary)]
        public async Task ExplicitTupleNameAdded_DeconstructionDeclaration2()
        {
            var code = @"
class C
{
    static int y = 1;
    void M()
    {
        int [||]i = C.y;
        var t = ((i, _) = (1, 2));
    }
}";

            var expected = @"
class C
{
    static int y = 1;
    void M()
    {
        var t = (((int)C.y, _) = (1, 2));
    }
}";
            await TestInRegularAndScriptAsync(code, expected, ignoreTrivia: false);
        }
```

**Comment by jcouv** [comment]
I won't be able to get to this in dev15.3.
